src/Compilers/VisualBasic/Portable/Compilation/VisualBasicCompilation.vb

```diff
@@ -2594,11 +2594,17 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
         End Function
 
         Protected Overrides Function CommonCreateTupleTypeSymbol(elementTypes As ImmutableArray(Of ITypeSymbol), elementNames As ImmutableArray(Of String)) As INamedTypeSymbol
-            Throw New NotSupportedException(VBResources.TuplesNotSupported)
+            Dim vbElementTypes = elementTypes.SelectAsArray(Function(t) t.EnsureVbSymbolOrNothing(Of TypeSymbol)(NameOf(elementTypes)))
```

Comment:
I think there should be additional validation on `elementNames`. It should be default or have the same number of elements as `elementTypes`. 
Also, with the new "partial names" scenario, we only care that all the names aren't null, if you specified any.